pkg/scheduler/reconciliation/persistent.go

```diff
@@ -447,3 +447,135 @@ func (r *PersistentReconciliationRepository) UpdateOperationRetryID(schedulingID
 	}
 	return db.Transaction(r.Conn, dbOps, r.Logger)
 }
+
+func (r *PersistentReconciliationRepository) UpdateOperationPickedUp(schedulingID, correlationID string) error {
+	dbOps := func(tx *db.TxConnection) error {
+		op, err := r.GetOperation(schedulingID, correlationID)
```

Comment:
Here, you should use the repository with the `tx` parameter like this:
```
rTx := r.WithTx(tx) 
rTx.GetOperation(...)
```

Reply:
Done! Also I need to use the `tx` instead of `r.Conn` 👍🏻